src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package mate.academy.service.impl;
+
+import java.util.Collections;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private MovieSessionDao movieSessionDao;
+    @Inject
+    private TicketDao ticketDao;
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket newTicket = ticketDao.add(new Ticket(movieSession, user));
+        ShoppingCart usersShoppingCart = shoppingCartDao.getByUser(user).orElseThrow(() ->
+                new RuntimeException("Can't add ticket: " + newTicket
+                        + ". There is no available Shopping cards at user: " + user));
```

Comment:
```suggestion
        ShoppingCart usersShoppingCart = getByUser(user);
```

I think we can use here `getByUser(User user)` method from this service (same in `clear()` method)